mobilizon_reshare/main/start.py

```diff
@@ -31,29 +44,24 @@ async def start():
     # Pull unpublished events from Mobilizon
     unpublished_events = get_unpublished_events(published_events)
     # Store in the DB only the ones we didn't know about
-    await create_unpublished_events(unpublished_events)
+    db_unpublished_events = await create_unpublished_events(unpublished_events)
     event = select_event_to_publish(
         published_events,
         # We must load unpublished events from DB since it contains
         # merged state between Mobilizon and previous WAITING events.
-        list(await get_db_unpublished_events()),
+        db_unpublished_events,
     )
 
     if event:
         logger.debug(f"Event to publish found: {event.name}")
 
-        waiting_publications_models = await publications_with_status(
-            status=PublicationStatus.WAITING, event_mobilizon_id=event.mobilizon_id,
-        )
-        waiting_publications = list(
-            map(
-                partial(EventPublication.from_orm, event=event),
-                waiting_publications_models.values(),
-            )
-        )
-
-        reports = PublisherCoordinator(waiting_publications).run()
+        models = await create_event_publication_models(event)
+        publications = list(EventPublication.from_orm(m, event) for m in models)
```

Comment:
Do we really need to do this here? I think that we should try and expose a "public" api that only knows about `MobilizonEvent`s and `EventPublication`s and then have a lower level that implements everything with models. maybe it could go in `mobilizon_reshare/storage/query/__init__.py` .

We already have a [function](https://github.com/Tech-Workers-Coalition-Italia/mobilizon-reshare/blob/c0706c3d8d910971758b7ae74c444a54b0fdaebd/mobilizon_reshare/storage/query/read_query.py#L116) that outputs a `dict[UUID, Publication]` and we always call `MobilizonEvent.from_model` before returning a list of events, so we could just return a `dict[UUID, EventPublication]` and have some consistency for publications as well.

Reply:
We can, I just don't like returning dicts in an API. I can make a container class for the publications but I tried to actively remove all the dicts and tuples and weird data structures we return, because they put too much effort on the reader and we easily end up with unnecessary complexity